pkg/storage/tsdb/caching_config.go

```diff
@@ -130,14 +134,18 @@ func CreateCachingBucket(chunksCache cache.Cache, chunksConfig ChunksCacheConfig
 			attributesCache = metadataCache
 		}
 		if chunksConfig.AttributesInMemoryMaxItems > 0 {
-			var err error
 			attributesCache, err = cache.WrapWithLRUCache(attributesCache, "chunks-attributes-cache", prometheus.WrapRegistererWithPrefix("cortex_", reg), chunksConfig.AttributesInMemoryMaxItems, chunksConfig.AttributesTTL)
 			if err != nil {
 				return nil, errors.Wrapf(err, "wrap metadata cache with in-memory cache")
 			}
 		}
-
-		cfg.CacheGetRange("chunks", chunksCache, isTSDBChunkFile, subrangeSize, attributesCache, chunksConfig.AttributesTTL, chunksConfig.SubrangeTTL, chunksConfig.MaxGetRangeRequests)
+		if chunksConfig.FineGrainedChunksCachingEnabled {
+			// When fine-grained caching is enabled, we are caching the chunks ourselves instead of relying on the caching bucket.
+			// In that case we will want to cache the attributes of chunk objects, but not their contents.
+			cfg.CacheAttributes("chunks-attributes", attributesCache, isTSDBChunkFile, chunksConfig.AttributesTTL)
```

Comment:
I think I told you to do it in a previous review, but now I think it's not necessary anymore. Attributes were read by `CachingBucket.cachedGetRange()` for each get range request. But given `cachedGetRange()` won't be used anymore, I can't see any other store-gateway use case that would have used the attributes caching (I checked both objstore clients implementation and direct calls to `Attributes()`).

Can you also double check it, please? Am I missing anything?

Reply:
no, i think you're right. The attributes cache that used to be enabled for the chunks config is only used when the caching bucket is also used

Reply:
I pushed a commit to not create an attributes cache when fine-grained chunks caching is enabled